source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -590,6 +590,66 @@ namespace
         }
 
         writeAttributes(node, destNode);
+
+        return destNode;
+    }
+
+    void writeMaterialElements(const PvtNode* node, NodePtr mxNode, DocumentPtr doc, const RtWriteOptions* writeOptions)
+    {
+        if (!writeOptions || writeOptions->materialElementType == RtWriteOptions::MaterialElementType::NONE ||
+            !(writeOptions->materialElementType & RtWriteOptions::MaterialElementType::WRITE))
+        {
+            return;
+        }
+
+        const PvtNodeDef* nodedef = node->getNodeDef()->asA<PvtNodeDef>();
+        const string type = nodedef->numOutputs() == 1
+                            ? nodedef->getPort(0)->getType().str()
+                            : "multioutput";
```

Comment:
Since this type string is only used for the if-statement test below, you don't need to involve the "multioutput" string. And you can compare the RtToken version of the type directly which is much faster: 
`if (nodedef->numOutputs() == 1 && nodedef->getPort(0)->getType() ==RtType::SURFACESHADER)`

Reply:
I would also more this test outside this function, to avoid calling this function on _all_ nodes we write. The test above on the writeOptions I would also move outside to the calling side to avoid calling the function when writeOptions says not to write.